### PR TITLE
Add a script to launch latest s0 AMI

### PR DESCRIPTION
**Pull request by e-kov**
Implemented a script to launch an EC2 instance from the Elastio s0 AMI.

This script has parameters:
1. a security group name of an existing security group with the open ports 22, 61234.
2. an instance role which can access to a bucket and a KMS key.
3. a bucket name.
4. KMS key alias.
5. shard type (fixed or variable).

This script uses cloud-init to:
- Format the locally attached NVMe drive.
- Run once `s0 vault create s3 --bucket <s0 bucket> --key-id <s0 kms key>` to create an s0 vault in the specified s3 bucket.
- Register a systemctl service and run it with the command inside like this `s0 vault serve s3 --bucket <s0 bucket> --key-id <s0 kms key> --fixed (or variable) --server-address 0.0.0.0`

Note: The bootstrap yaml is in the same script as heredoc for the user's convenience. It's enough for them to download just one file.

Resolves: elastio/packaging#40
Resolves: elastio/elastio#1326

### CODE REVIEW

File: scripts/launch-s0-instance.sh [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+elastio_aws_id=537513441174
+security_group="elastio-s0-server"
+default_instance_type="m5d.2xlarge"
+default_instance_name="$(whoami)-s0-server"
+
+usage()
+{
+    echo "Usage examples:"
+    echo "   $me --key-name my_ec2_key --name my_s0_server --type m5d.2xlarge"
+    echo "   $me --k /path/to/key.pem -n my_s0_server -t m5d.2xlarge"
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:14]: This is misleading.  the key name isn't a path on the filesystem, it's the name assigned to the key pair in the EC2 console

File: scripts/launch-s0-instance.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+fi
+
+[ -z "$instance_name" ] && instance_name=$default_instance_name
+[ -z "$instance_type" ] && instance_type=$default_instance_type
+
+latest_ami=$(aws ec2 describe-images --owners $elastio_aws_id \
+    --filters "Name=name,Values=s0-ami*" "Name=state,Values=available" \
+    --query "reverse(sort_by(Images, &CreationDate))[:1].ImageId" --output text)
+
+if aws ec2 describe-security-groups | jq '.SecurityGroups[].GroupName' | tr -d '"' | grep -q "$security_group" ; then
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:57]: That's a great convenience feature but it should be possible to override this and specify a security group of the caller's choosing.

File: scripts/launch-s0-instance.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+echo "Launched EC2 instance id $instance_id"
+
+echo "Querying instance info for public DNS..."
+instance_info=$(aws ec2 describe-instances --instance-ids $instance_id --output json)
+
+instance_dns=$(echo $instance_info | jq ".Reservations[].Instances[].PublicDnsName" -r)
+
+echo "The instance DNS is $instance_dns"
+echo
+echo "SSH into the instance using the key '$key_name' like this: 'ssh -i /path/to/$key_name.pem ec2-user@$instance_dns'"
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:95]: That path expression is also misleading.

File: scripts/launch-s0-instance.sh [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    if ! aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 22 --cidr 0.0.0.0/0 ||
+       ! aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 61234 --cidr 0.0.0.0/0 ; then
+        echo "Failed to open ports 22 and 61234 for s0 server and ssh!"
+        exit 3
+    fi
+fi
+
+echo "Launching \"$instance_name\" (instance type $instance_type) with the latest Elastio s0 AMI ($latest_ami)"
+
+instance_json=$(aws ec2 run-instances \
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:73]: The `launch-instance.sh` script in `elastio/elastio` also uses cloud-init to format the locally attached NVMe drive and install some packages.  Obviously the package install stuff doesn't make sense here since this is our AMI, but setting up the NVMe disk and mounting it somewhere would also be a valuable addition.

On the call today we also talked about scripting the rest of the process, which is launching `s0` itself and serving a vault given the bucket name, KMS key alias, and shard type (fixed or variable).    Do you plan to do that in a separate script?  It would be a better user experience if they're together.

**Review by anelson — CHANGES_REQUESTED** [review]

File: scripts/launch-s0-instance.sh (hunk 2 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:57]: Do you mean add an optional script parameter for a security group name and create a group with this name if it doesn't exist? Or just use a customer's existing group? I'm sure no one has a group with the 61234 port opened.

File: scripts/launch-s0-instance.sh (hunk 3 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:95]: Why? Here isn't a key name, but a path. And param `-i` is necessary.

File: scripts/launch-s0-instance.sh (hunk 4 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:73]: Ok, what size that disk should be? And why we need a disk to add? Does s0 use an s3 bucket and doesn't use a local storage?

I've started to work on this script in terms of https://github.com/elastio/packaging/issues/40 before today's decision to make it extended with the `s0` configuration. And there is a new issue created after the discussion by @mvovchuk https://github.com/elastio/elastio/issues/1326 which have to be discussed. For now I have no idea how s0 should be configured.
My preference to get a script which just starts an instance as of https://github.com/elastio/packaging/issues/40 and then extend it to config s0 as of https://github.com/elastio/elastio/issues/1326. I plan to have all of it in this script, in two steps.

File: scripts/launch-s0-instance.sh (hunk 1 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: You are right. There is a kind of key name in the example with the long keys line above. Fixed.

File: scripts/launch-s0-instance.sh [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+[ "$1" == "" ] && echo "Script need some arguments!" && usage && exit -1
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -k | --key-name)   shift && key_name=$1 ;;
+        -n | --name)       shift && instance_name=$1 ;;
+        -t | --type)       shift && instance_type=$1 ;;
+        -h | --help)       usage && exit ;;
```
Comment (e-kov) [inline scripts/launch-s0-instance.sh:34]: Add a parameter for an instance role which can access to a bucket and a KMS key

File: scripts/launch-s0-instance.sh (hunk 4 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:223]: done, see the list here https://github.com/elastio/elastio/issues/1326

File: scripts/launch-s0-instance.sh (hunk 2 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: done

File: scripts/launch-s0-instance.sh (hunk 3 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: simplified the message, copy-pasted your one from [here](https://github.com/elastio/elastio/blob/280ee677fe731c1ee594b93d247e618c8e984a44/scripts/launch-instance.sh#L90).

File: scripts/launch-s0-instance.sh (hunk 5 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: done.

File: scripts/launch-s0-instance.sh [hunk 6]
```diff
@@ -3,81 +3,233 @@
...
+      parted -a optimal --script /dev/nvme1n1 mkpart primary 0% 100%
+      mkfs.xfs -f /dev/nvme1n1p1
+      mount /dev/nvme1n1p1 $DB_MOUNT
+    fi
+    chown -R ec2-user:ec2-user $DB_MOUNT
+
+  # Configure AWS region
+  - |
+    echo "region = \$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | grep -w region | cut -d: -f2 | tr -d '", ')" >> /home/ec2-user/.aws/config
+    cp -r /home/ec2-user/.aws /root/
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:71]: This shouldn't be necessary with `s0` since it doesn't ever need to run as `root`, but it will be necessary in a later automation that launches ingest VMs...

File: scripts/launch-s0-instance.sh [hunk 7]
```diff
@@ -3,81 +3,233 @@
...
+
+  # Create s0 vault
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:73]: Vault creation and vault serving are separate functions.  The way this is written, every time you launch an instance you will re-create the vault, overwriting any data from a previous run.  That's obviously not particularly useful.

Automating vault creation is now less important than automating vault serving, because Aleks is almost done with elastio/elastio#1181.

File: scripts/launch-s0-instance.sh [hunk 8]
```diff
@@ -3,81 +3,233 @@
...
+  # Prepare and start s0 service
+  - |
+    touch /var/log/s0-service.log
+    chown ec2-user:ec2-user /var/log/s0-service.log
+    systemctl daemon-reload
+    systemctl enable $UNIT_FILE
+    systemctl start $SERVICE
+
+output:
+  all: '| tee -a /var/log/cloud-init-output.log'
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:86]: Why is this needed?  I always `tail -f /var/log/clout-init-output.log` to monitor the progress of cloud init tasks, and I never out this `output` element in my cloud init scripts.

File: scripts/launch-s0-instance.sh [hunk 9]
```diff
@@ -3,81 +3,233 @@
...
+      [Unit]
+      Description="Elastio Scale-to-Zero service"
+      After=network.target
+
+      [Service]
+      User=ec2-user
+      StartLimitInterval=5
+      StartLimitBurst=10
+      TimeoutStopSec=5min
+      ExecStart=/bin/bash -c '/usr/bin/s0 --log-output compact vault serve s3 --bucket $S0_BUCKET --key-id $S0_KMS $BLOCK_SIZE --server-address 0.0.0.0 --db-path $DB_MOUNT 2>&1 | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g" >> /var/log/s0-service.log'
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:43]: the `key-id` parameter can be either the ugly AWS-generated key ID, or a human-readable alias.  If it's the latter, it needs the `alias/` prefix otherwise it will not be valid.  It would be great if you could do some minimal validation on the key ID before launching to avoid this failure mode.

Comment (anelson) [inline scripts/launch-s0-instance.sh:43]: If either the bucket or KMS key do not exist in the region, this will fail.  Can you use the AWS CLI to validate both of these before launch?

File: scripts/launch-s0-instance.sh [hunk 10]
```diff
@@ -3,81 +3,233 @@
...
-    echo "  -k | --key-name : A name of an aws EC2 key in the PEM format for the SSH connection to the instance."
+    echo "  -b | --bucket-name        : s3 bucket name for s0 vault."
     echo
-    echo "  -n | --name     : Optional. Name of an instance. It's shown in the AWS console."
-    echo "                    The default value is \"[user_name]-s0-server\"."
+    echo "  -k | --kms-key-alias      : KMS key alias for the data encription in the s0 vault."
     echo
-    echo "  -t | --type     : Optional. An instance type. See https://aws.amazon.com/ec2/instance-types/ for more details."
-    echo "                    The default value is \"m5d.2xlarge\"."
+    echo "  -p | --instance-profile   : An instance provile with access to the s3 bucket and KMS key from 2 parametrers abowe."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:114]:
```suggestion
    echo "  -p | --instance-profile   : An instance provile with access to the s3 bucket and KMS key from 2 parameters above."
```

File: scripts/launch-s0-instance.sh [hunk 11]
```diff
@@ -3,81 +3,233 @@
...
+    echo "                              See aws doscs for more details how to create one:"
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:115]:
```suggestion
    echo "                              See AWS docs for more details how to create one:"
```

File: scripts/launch-s0-instance.sh [hunk 12]
```diff
@@ -3,81 +3,233 @@
...
+    echo "  -s | --ssh-key-name       : A name of an aws EC2 key in the PEM format for the SSH connection to the instance."
+    echo
+    echo "  -n | --instance-name      : Optional. Name of an instance. It's shown in the AWS console."
+    echo "                              The default value is \"[user_name]-s0-server\"."
+    echo
+    echo "  -t | --instance-type      : Optional. An instance type. See https://aws.amazon.com/ec2/instance-types/ for more details."
+    echo "                              The default value is \"m5d.2xlarge\"."
+    echo
+    echo "  -g | --security-group     : Optional. A name of the security group which has ports 22 and 61234 open to the world for SSH and s0 server respectively."
+    echo "                              The grop 'elastio-s0-server' will be created and then used in case if this OR next argument is not specified."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:106]:
```suggestion
    echo "                              The group 'elastio-s0-server' will be created and then used in case if this OR next argument is not specified."
```

File: scripts/launch-s0-instance.sh [hunk 13]
```diff
@@ -3,81 +3,233 @@
...
+    echo "                              NOTE: The group existance and the pen ports in the group aren't checked."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:107]:
```suggestion
    echo "                              NOTE: The group existance and the open ports in the group aren't checked."
```

File: scripts/launch-s0-instance.sh [hunk 14]
```diff
@@ -3,81 +3,233 @@
...
+    echo "                                    The script may not have permissions and assumes that the group is configured properly."
     echo
-    echo "  -k | --key-name : A name of an aws EC2 key in the PEM format for the SSH connection to the instance."
+    echo "  -b | --bucket-name        : s3 bucket name for s0 vault."
     echo
-    echo "  -n | --name     : Optional. Name of an instance. It's shown in the AWS console."
-    echo "                    The default value is \"[user_name]-s0-server\"."
+    echo "  -k | --kms-key-alias      : KMS key alias for the data encription in the s0 vault."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:112]: I didn't notice before I made my other command that this parameter is explicitly the key **alias**, not the key ID.  Then you can prepend the `alias/` prefix automatically, if the user hasn't already prepended it themselves.

File: scripts/launch-s0-instance.sh [hunk 15]
```diff
@@ -3,81 +3,233 @@
...
     echo
-    echo "  -t | --type     : Optional. An instance type. See https://aws.amazon.com/ec2/instance-types/ for more details."
-    echo "                    The default value is \"m5d.2xlarge\"."
+    echo "  -p | --instance-profile   : An instance provile with access to the s3 bucket and KMS key from 2 parametrers abowe."
+    echo "                              See aws doscs for more details how to create one:"
+    echo "                              https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html"
+    echo "                              https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html"
     echo
-    echo "  -h | --help     : Show this usage help."
+    echo "  -z | --s0-block-size      : Optional. A block size for the s0 shard. There are 2 possible values: \"fixed\" and \"variable\". The \"fixed\" is default."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:119]:
```suggestion
    echo "  -z | --shard      : Optional. Which of the vault's shards to serve. There are 2 possible values: \"fixed\" and \"variable\". The \"fixed\" is default."
```

File: scripts/launch-s0-instance.sh [hunk 16]
```diff
@@ -3,81 +3,233 @@
...
+    echo "                              - The \"fixed\" selects the shard with a fixed-length block size."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:120]:
```suggestion
    echo "                              - The \"fixed\" selects the shard with using fixed-block deduplication."
```

File: scripts/launch-s0-instance.sh [hunk 17]
```diff
@@ -3,81 +3,233 @@
...
+    echo "                                This shard is intended for use with backups of block storage systems like EBS, local disks, and VMs."
+    echo "                              - The \"variable\" selects the shard with a variable-length block size."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:122]:
```suggestion
    echo "                              - The \"variable\" selects the shard using variable (i.e. content-defined) deduplication."
```

File: scripts/launch-s0-instance.sh [hunk 18]
```diff
@@ -3,81 +3,233 @@
...
+        -h | --help)                usage && exit ;;
+        *)                          echo "Wrong arguments!"
+                                    usage && exit 15 ;;
     esac
         shift
 done
 
+if ! which aws >/dev/null 2>&1 || [[ $(aws --version | cut -d' ' -f1 | cut -d'/' -f2 | cut -d'.' -f1) < "2" ]]; then
+    echo "Please install aws cli v.2 and try again."
+    echo "You can use a package manager i.e. dnf/yum or apt or follow this instruction https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:150]:
```suggestion
    echo "You can use a package manager i.e. dnf/yum or apt or follow these instructions https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html."
```

File: scripts/launch-s0-instance.sh [hunk 19]
```diff
@@ -3,81 +3,233 @@
...
+        -z | --s0-block-size)       shift && block_size=$1 ;;
+        -h | --help)                usage && exit ;;
+        *)                          echo "Wrong arguments!"
+                                    usage && exit 15 ;;
     esac
         shift
 done
 
+if ! which aws >/dev/null 2>&1 || [[ $(aws --version | cut -d' ' -f1 | cut -d'/' -f2 | cut -d'.' -f1) < "2" ]]; then
+    echo "Please install aws cli v.2 and try again."
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:149]:
```suggestion
    echo "Please install AWS CLI v2 and try again."
```

File: scripts/launch-s0-instance.sh [hunk 20]
```diff
@@ -3,81 +3,233 @@
...
-        echo "Failed to create a security group to open ports for s0 server and ssh. Do you have enough permissions?"
-        exit 2
-    fi
-    if ! aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 22 --cidr 0.0.0.0/0 ||
-       ! aws ec2 authorize-security-group-ingress --group-id $sg_id --protocol tcp --port 61234 --cidr 0.0.0.0/0 ; then
-        echo "Failed to open ports 22 and 61234 for s0 server and ssh!"
-        exit 3
+if [ -z "$security_group" ]; then
+    security_group=$default_security_group
+    if seq_gropups=$(aws ec2 describe-security-groups); then
```
Comment (anelson) [inline scripts/launch-s0-instance.sh:190]:
```suggestion
    if seq_groups=$(aws ec2 describe-security-groups); then
```

**Review by anelson — CHANGES_REQUESTED** [review]

File: scripts/launch-s0-instance.sh (hunk 6 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:71]: Right now it's already necessary before the call `s0 vault create s3 ...` below. These cloud-init instructions in the `runcmd:` are running as `root`.

File: scripts/launch-s0-instance.sh (hunk 7 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:73]: Will https://github.com/elastio/elastio/pull/1281 be landed before the end of sprint?
What is the new expected behaviour? Will `s0 vault serve s3  ...` create a vault if it doesn't exist? Or is it possible to check, does it exist by some command and create it or not according to the check result?

For now I just know, that if a bucket is empty and a vault doesn't exist, then the service with the command `s0 vault serve s3  ...` will fail with an error and probably just exit returning zero )

File: scripts/launch-s0-instance.sh (hunk 8 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:86]: I found this in the AWS doc https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html and used for debugging to resolve issues of the bootstrap scripts. However I found very similar log file inside of VM `/var/log/cloud-init.log`. Probably you are referring to it. But `/var/log/cloud-init-output.log` doesn't exist by default.
If you like, I can remove it because debugging is done.

File: scripts/launch-s0-instance.sh (hunk 9 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:43]: The validation makes sense. But I always used an alias without the mentioned prefix `alias/` and this command wasn't failing.
Where to put this prefix? Please explain how to use it. Do you mean something like this?
`--key-id alias/my-kms-key` ?

File: scripts/launch-s0-instance.sh (hunk 16 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:120]: Maybe `with` is excess? Just `The \"fixed\" selects the shard using fixed-block deduplication.` ?

File: scripts/launch-s0-instance.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: Visually it looks like the `compact` just use less indents and one message is a one-line instead of multiple lines.
Removed this argument. By default it's `full`.

File: scripts/launch-s0-instance.sh (hunk 7 above)
Comment (anelson) [reply scripts/launch-s0-instance.sh:73]: I don't have a high confidence that Lambda-powered vault creation will land this sprint, no.  But this behavior makes launching the instance useful only as a way to see if ingests are working.  As soon as the instance is launched again, the vault is re-created and all backups are lost.  That's not what we're going for.  I'd rather the user have to manually run `s0 vault create s3 ...` than have to explain this behavior.

BTW the fix for #1375 is awaiting your review )

File: scripts/launch-s0-instance.sh (hunk 8 above)
Comment (anelson) [reply scripts/launch-s0-instance.sh:86]: Strange.  I've always used `cloud-init-output.log` and I never used a directive like this to create it.  I wonder what the difference is.

Anyway it's not hurting anything.  That's just an FYI.

File: scripts/launch-s0-instance.sh (hunk 9 above)
Comment (anelson) [reply scripts/launch-s0-instance.sh]: It wasn't failing because of elastio/elastio#1327 but it would fail at runtime for sure.

Yes that is what I mean.  If you know that the parameter is always going to be an alias and never an actual key ID (and from the usage I'd say you know that), then you should verify it starts with `alias/` and if it doesn't you should prepend that prefix.

**Review by anelson — DISMISSED** [review]

File: scripts/launch-s0-instance.sh (hunk 6 above)
Comment (anelson) [reply scripts/launch-s0-instance.sh:71]: That seems wrong.  We've discussed elsewhere why I don't think you should unconditionally `create` at all, but to do it as root...why?  You can `su` to run as the non-root user, although in this case I think the better fix is to not create the vault at all.

File: scripts/launch-s0-instance.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (anelson) [reply scripts/launch-s0-instance.sh]: It's more than that.  The compact format doesn't display all span fields; just the most recent.

File: scripts/launch-s0-instance.sh (hunk 7 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:73]: Yup, I agree that the overwriting of the vault on each start is a bad idea. And this behaviour should be changed.
But your suggestion to run manually `s0 vault create s3 ...` buy user is bad idea because it means that they'll need to start s0 service manually then...
I can do the next:
Check existence of the file `vault.json` in the bucket. If it exists, the vault is already created and I don't need to create it here. Otherwise `s0 vault create s3 ...` is running here.
Does it makes sense?

File: scripts/launch-s0-instance.sh (hunk 9 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: Thanks for the response. It makes sense.

File: scripts/launch-s0-instance.sh (hunk 6 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:71]: See https://github.com/elastio/elastio-stack/pull/3#discussion_r560887316

File: scripts/launch-s0-instance.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (e-kov) [reply scripts/launch-s0-instance.sh]: Good to know!

File: scripts/launch-s0-instance.sh (hunk 6 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:71]: yes, I can `su`. But an instance is in some region despite the user you use inside of it. )) Why don't do this simplification and don't worry in the future that hypothetically some command can be executed as sudo or root and fail because we are decided to not do this simplification? ))

File: scripts/launch-s0-instance.sh (hunk 7 above)
Comment (anelson) [reply scripts/launch-s0-instance.sh:73]: Agreed!

**Review by anelson — COMMENTED** [review]
I'm just awaiting the enhancement to auto create a vault if one isn't present and I'm ready to approve this.

File: scripts/launch-s0-instance.sh (hunk 14 above)
Comment (e-kov) [reply scripts/launch-s0-instance.sh:117]: Done. See this change https://github.com/elastio/elastio-stack/pull/3/commits/44eecd6414d4eafca88bb895083e6c982871f696#diff-fbbd938ca5f495c8ec4d098678874f45915efb2537ed6ad329ff3fc5c4f98845R43
and this change https://github.com/elastio/elastio-stack/pull/3/commits/44eecd6414d4eafca88bb895083e6c982871f696#diff-fbbd938ca5f495c8ec4d098678874f45915efb2537ed6ad329ff3fc5c4f98845R143

**Review by anelson — APPROVED** [review]
